mono/metadata/threads.c

```diff
@@ -784,6 +784,10 @@ mono_thread_internal_set_priority (MonoInternalThread *internal, MonoThreadPrior
 #else
 	min = sched_get_priority_min (policy);
 	max = sched_get_priority_max (policy);
+
+	/* Not tunable. Bail out */
+	if ((min == -1) || (max == -1))
+		return;
 #endif
 	MONO_EXIT_GC_SAFE;
```

Comment:
We need to ensure we do the MONO_EXIT_GC_SAFE even if we return from the `-1` case.

I'd just move the checks here outside the #if, assuming PRIORITY_MIN isn't -1 on `__PASE__` which is probably a safe bet.